src/frontend/src/planner/query.rs

```diff
@@ -54,7 +54,7 @@ impl Planner {
         }
         let mut out_fields = FixedBitSet::with_capacity(plan.schema().len());
         out_fields.insert_range(..plan.schema().len() - extra_order_exprs_len);
-        if plan.as_logical_project_set().is_some() {
+        if let Some(field) = plan.schema().fields.get(0) && field.name == "projected_row_id"  {
```

Comment:
Do we always assume that the `project_set` is the top level operator?

Reply:
We did, and this PR fixed that. See the planner test above, where `project_set` is not at the top level, but output column contains `projected_row_id`.

Reply:
So we are loosing the assumption to this: `projected_row_id` is always the 0-th column / there are no PlanNode after ProjectSet that can reorder the schema.